src/Microsoft.Azure.WebJobs.Host/Bindings/Path/BindingTemplateToken.cs

```diff
@@ -1,32 +1,271 @@
+                builder.Append(String.Format(CultureInfo.InvariantCulture, "(?<{0}>.*)", value));
+            }
+
+            protected virtual string Fallback()
+            {
+                // Derived class can provide fallback resolutions if we don't match binding data. 
+                return null;
+            }
+
+            public override string Evaluate(IReadOnlyDictionary<string, object> bindingData)
+            {
+                object current;
+
+                if (bindingData.TryGetValue(this._expressionParts[0], out current))
+                {
+                    for (int i = 1; i < _expressionParts.Length; i++)
+                    {
+                        var propName = _expressionParts[i];
+
+                        try
+                        {
+                            current = GetProperty(current, propName);
+                        }
+                        catch (Exception e)
+                        {
+                            throw new InvalidOperationException($"Error while accessing '{propName}': {e.Message}");
+                        }
+                    }
+                }
+                else
+                {
+                    // Not found. Maybe a resolver can do it?
+                    current = this.Fallback();
+
+                    if (current == null)
+                    {
+                        throw new InvalidOperationException(String.Format(CultureInfo.CurrentCulture, "No value for named parameter '{0}'.", this.ParameterName));
+                    }
+                }
+
+                var strValue = BindingDataPathHelper.ConvertParameterValueToString(current);
+                return strValue;
+            }
+
+            // Case insensitive lookup.
+            // Helper to get a property from an object.             
+            // This supports both static types and JObject (for dynamic) 
+            public static object GetProperty(object o, string member)
+            {
+                Type scope = o.GetType();
+                JObject jobj = o as JObject;
+                if (jobj != null)
+                {
+                    JToken propValue;
+                    // JObject's normal dot operator returns null for missing properties; 
+                    // we need to explicitly call TryGetValue.
+                    if (!jobj.TryGetValue(member, StringComparison.OrdinalIgnoreCase, out propValue))
+                    {
+                        throw new InvalidOperationException($"property doesn't exist.");
+                    }
+                    return propValue;
+                }
+                else
+                {
+                    var prop = scope.GetProperty(member, BindingFlags.Public | BindingFlags.Instance | BindingFlags.IgnoreCase);
+                    if (prop == null)
+                    {
+                        throw new InvalidOperationException($"property doesn't exist.");
```

Comment:
these will get wrapper by the caller and it will append additional information like the method name. Is that what you had in mind?

---
In reply to: [114872462](https://github.com/Azure/azure-webjobs-sdk/pull/1117#discussion_r114872462) [](ancestors = 114872462)